File: src/frontend/test/src/domain/entities/Adresse.ts
import type { ILogement } from './Logement';
import type { IUtilisateur } from '@/domain/entities/Utilisateur';

export interface IAdresse {
    id: number;
    nomAdresse: string;
    complementAdresse?: string;
    ville: string;
    codePostal: number;
    pays: string;
    logements?: ILogement[];
    utilisateurs?: IUtilisateur[];
}

export class Adresse implements IAdresse {
    constructor(
        public id: number,
        public nomAdresse: string,
        public ville: string,
        public codePostal: number,
        public pays: string,
        public complementAdresse?: string,
        public logements?: ILogement[],
        public utilisateurs?: IUtilisateur[]
    ) {}
}

File: src/backend/NHP_Domain/Entities/Adresse.cs
namespace NHP_Domain.Entities;

public class Adresse
{
    public long Id { get; set; }
    public required String NomAdresse { get; set; }
    public String? ComplementAdresse { get; set; }
    public required String Ville  { get; set; }
    public required int CodePostal { get; set; }
    public required String Pays { get; set; }
    
    // OneToMany
    //public virtual List<Logement>? Logements { get; set; }
    //public virtual List<Utilisateur>? Utilisateurs { get; set; }

    public override string ToString()
    {
        return $"NomAdresse: {NomAdresse} " +
               $"complementAdresse: {ComplementAdresse} " +
               $"ville: {Ville} " +
               $"codePostal: {CodePostal} " +
               $"pays: {Pays} ";
    }
}